### DISCUSSION TITLE
Live Research Progress Tracking

### DISCUSSION THREAD

**Discussion by Skilly146** [Tsangares/stoned_ape_tools#16] [Feature Requests]
## What 

Make the extension able to identify its own 'share research' formatted messages and replace it with a live update of where the progress should be that tick and when it will finish

## How

The easiest way would be to multiply ticks since the message was sent by the senders current science to figure out how much progress they have made since then. The problem with this though is that it doesn't account for if the sender acquires more science. We could mitigate this a little bit by storing how much science the sender has every time the user logs on as well as using the intel screen to make a more accurate calculation. That still wouldn't be perfect but worst case it says the tech finishes later than it actually does.

**Comment by anicolao** [comment]
I want to add a different but perhaps related feature. 

[[Tick #]] should be displayed in the user's current chosen time format (cycled with the `%` key). So if I write a message saying: 
```
Let's meet at [[Enemy Planet]] [[Tick 45]]
```

the recipient would see one of 
```
Let's meet at Enemy Planet @ 12:02PM.
```
if their time is set to absolute, or
```
Let's meet at Enemy Planet 3h32m42s.
```
if their time is set to relative. 

If this was implemented, the research paste could just paste tick #s instead of computing ticks remaining.

As an aside, there is no reason why ticks remaining shouldn't be a time format, so you could see all times as "in N ticks".
